Basecamp_Project_Tests/WebApp.Tests/PointsControllerTest.cs

```diff
@@ -0,0 +1,168 @@
+﻿using BaseCamp;
+using FluentAssertions;
+using Microsoft.AspNetCore.Mvc;
+using Moq;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using WebApp.Api.Requests;
+using WebApp.Api.Responses;
+using WebApp.Core.Models;
+using WebApp.Core.Repositories;
+using Xunit;
+
+namespace WebApp.Tests
+{
+    public class PointsControllerTest
+    {
+        [Fact]
+        public void GetAllPoints_returnsActionResultOkWithAllPoints()
+        {
+            // Arrange
+            var mock = new Mock<IRepository<Point>>();
+            mock.Setup(repo => repo.GetAll()).Returns(GetTestPoints());
+            var controller = new PointsController(mock.Object);
+            // Act
+            var result = controller.GetAllPoints();
+            // Assert
+            result.Should().BeOfType<OkObjectResult>();
+            result.As<OkObjectResult>().Value.As<IEnumerable<PointResponse>>().Should().HaveCount(GetTestPoints().Count);
+            result.As<OkObjectResult>().Value.As<IEnumerable<PointResponse>>().Should().BeEquivalentTo(GetTestPoints().Select(p => new PointResponse(p)));
+        }
+
+        [Fact]
+        public void GetPointById_whenSpecifiedPointExists_returnsOkResultWithSpecifiedPoint()
+        {
+            // Arrange
+            int testPointId = 1;
+            var mock = new Mock<IRepository<Point>>();
```

Comment:
You can create a field of type Mock<IRepository<Point>> and initialize in the ctor